tools/lib/uobjnew.py

```diff
@@ -98,7 +98,20 @@
     u64 classptr = 0, size = 0;
     bpf_usdt_readarg(2, ctx, &classptr);
     bpf_usdt_readarg(4, ctx, &size);
-    bpf_probe_read_user(&key.name, sizeof(key.name), (void *)classptr);
+    // The reason why read the third arg twice with different types here:
+    // According to the source code of OpenJDK:
+    // the declaration type of the third arg is uintptr_t(see hotspot.d)
+    // but the actual type is int(see SharedRuntime::dtrace_object_alloc)
+    u32 len32 = 0;
+    u64 len64 = 0;
+    size_t len = sizeof(key.name);
+    if (!bpf_usdt_readarg(3, ctx, &len32)) {
+      len = min(len, (size_t)len32);
+    } else {
+      bpf_usdt_readarg(3, ctx, &len64);
```

Comment:
Do we have any chance to hit this branch?

Reply:
This branch is never hit for the JDK version(the mainline).
Not sure about other versions (8, 11 ...)

Reply:
Let us just use 'int' type, i.e., len32 version. It is unlikely it will change to len64. Also if the argument type is indeed like uintptr_t. The current implementation is not correct anyway since you will need to do an additional dereference to get the actual length when reading argument which is really unnecessary for usdt.

Reply:
okay, fixed.